include/mbedtls/ctr_drbg.h

```diff
@@ -30,6 +30,7 @@
 #endif
 
 #define MBEDTLS_ERR_CTR_DRBG_ENTROPY_SOURCE_FAILED        -0x0034  /**< The entropy source failed. */
+#define MBEDTLS_ERR_CTR_DRBG_AES_CRYPTO_FAILED                -0x0035  /**< Some of the crypto API (AES) has failed */
```

Comment:
The library has rather too many error codes as it is. Could we do without this one, and instead propagate the error from the underlying AES module? This is what other modules do, for example the cipher module propagates error from the underlying cipher and uses its own error codes only for errors detected in the cipher module itself.

Reply:
@gilles-peskine-arm - I think whatever error code we use here should follow the same naming convention as for hardware failures in other modules. eg. ` MBEDTLS_ERR_AES_HW_FAILED` possibly, if the error is going to be propagated from AES.

Reply:
After some internal discussion, we're going to add a standard error code `MBEDTLS_ERR_AES_HW_FAILED` to report errors from the AES hardware accelerator. In the CTR module, please don't use a separate error code, propagate that one instead.

Reply:
I've worked on this with Ron E. several weeks ago and we came  up with the above code.
Please discuss it with him.

Reply:
@NirEkhauz We haven't discussed the error code name.  Please change the name as suggested

Reply:
I will ignore it.
gilles-peskine-arm Please let me know when this new error code is in, so I'll be able to use it!

Reply:
The error code is `MBEDTLS_ERR_AES_HW_FAILED` as defined by https://github.com/ARMmbed/mbedtls/pull/1303 which we're planning to merge today (pending review and CI).

Reply:
Where it is going to be defined?
I need to know which file to include.

Reply:
It'll be defined in `aes.h`. But the code in `ctr_drbg.c` shouldn't use that code, it should just propagate whatever error the AES implementation returns.

Reply:
@gilles-peskine-arm I've pushed the change